### PR TITLE
[1] feat(1041): Pass manifest to checkoutConfig

### PR DESCRIPTION
**Pull request by Filbird**
Pending merge of https://github.com/screwdriver-cd/data-schema/pull/236

https://github.com/screwdriver-cd/screwdriver/issues/1041

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -163,6 +165,12 @@ class ScmBase {
             checkoutConfig.prRef = o.build.prRef;
         }
 
+        const manifest = getAnnotations(o.job.permutations[0], 'screwdriver.cd/repoManifest');
+
+        if (manifest) {
+            checkoutConfig.manifest = manifest;
```
Comment (minz1027) [inline index.js]: Should also update the jsdoc for getCheckoutCommand for the optional field `manifest`

File: index.js [hunk 2]
```diff
@@ -144,6 +145,7 @@ class ScmBase {
      * @method getSetupCommand
      * @param  {Object}         o           Information about the environment for setup
      * @param  {PipelineModel}  o.pipeline  Pipeline model for the build
+     * @param  {JobModel}       o.job       Job model for the build
```
Comment (minz1027) [inline index.js]: Maybe Object? since we put o.build as Object. Just to be consistent

File: index.js [hunk 3]
```diff
@@ -144,6 +146,7 @@ class ScmBase {
      * @method getSetupCommand
      * @param  {Object}         o           Information about the environment for setup
      * @param  {PipelineModel}  o.pipeline  Pipeline model for the build
+     * @param  {Object}       o.job         Job configuration for the build
```
Comment (tkyi) [inline index.js]: indentation is off

File: index.js [hunk 4]
```diff
@@ -163,6 +166,12 @@ class ScmBase {
             checkoutConfig.prRef = o.build.prRef;
         }
 
+        const manifest = getAnnotations(o.job.permutations[0], 'screwdriver.cd/repoManifest');
```
Comment (tkyi) [inline index.js]: Maybe define `screwdriver.cd/repoManifest` as a constant at the top of the file

File: README.md [hunk 5]
```diff
@@ -111,6 +111,7 @@ Required parameters:
 | config        | Object | Yes | Configuration Object |
 | config.branch | String | Yes | Pipeline branch |
 | config.host | String | Yes | Scm host (ex: github.com) |
+| config.manifest | Stirng | No | Repo Manifest URL |
```
Comment (tkyi) [inline README.md]: String*

File: README.md [hunk 6]
```diff
@@ -139,7 +140,8 @@ Required Parameters:
 | :-------------   | :---- | :--------| :---- |
 | config        | Object | Yes | Configuration Object |
 | config.pipeline | PipelineModel | Yes | Pipeline model |
-| config.build | Object | Yes | build config with sha and possibly prRef |
+| config.job | Object | yes |Job config with repoManifest annotation
```
Comment (tkyi) [inline README.md]: capital Yes

File: README.md (hunk 5 above)
Comment (Filbird) [reply README.md]: Oof x.x

**Pull request merged**